### PR TITLE
Add man page generation to Makefile

### LINKED ISSUE #123: add man pages
I think it's time to add man pages to vgrep. As many other go projects, we can use [go-md2man](https://github.com/cpuguy83/go-md2man) and use markdown. The content can be borrowed from the README.

### PR DESCRIPTION
**Pull request by ahmedkrmn**
Closes #123 

- Update `install` target to download and build `go-md2man` package.
- Add `docs` target to build man page from `README.md` using `go-md2man`.

### CODE REVIEW

**Review by vrothberg — COMMENTED** [review]
Thanks for tackling the issue! There are a couple of issues we need to address:

* The README is not a man page but we can recycle its content to create a new man page. Please have a closer look at Skopeo and its [Makefile](https://github.com/containers/skopeo/blob/master/Makefile). The man pages should be in a separate ./docs directory and be installed alongside vgrep via `make install`. Note that man pages cannot include links to images, so we cannot copy the README line-by-line. One `./docs/vgrep.1.md` should suffice.

* `make install` should not install `go-m2dman`. I suggest to create a dedicated make target for that (e.g., `.install.go-m2dman`).

Feel free to reach out if you any questions.

**Comment by ahmedkrmn** [comment]
Hi @vrothberg, 
Please have another look on this one.
I'm still unsure about the content of the `vgrep.1.md` file. Shall we reference some parts from the `grep` man?

File: Makefile [hunk 1]
```diff
@@ -25,6 +26,12 @@ export COVERAGE_PATH
 export COVERAGE_PROFILE
 $(shell mkdir -p ${COVERAGE_PATH})
 
+GOPATH ?= $(shell go env GOPATH)
+GOMD2MAN ?= $(GOPATH)/src/cpuguy83/go-md2man
```
Comment (vrothberg) [inline Makefile:30]: Tha path is missing `github.com`. I'd love to first check if `go-md2man` is already installed on the system. The Skopeo Makefile is a good example how to do that. A `go install github.com/cpuguy83/...` should be enough to install the binary in $GOBIN.

**Review by vrothberg — COMMENTED** [review]
The text looks good to me. Referencing grep, vgrep and rip-grep sounds like a good idea. Also, please have a look at the skopeo man page (https://raw.githubusercontent.com/containers/skopeo/master/docs/skopeo.1.md). We need the SYNOPSIS header and similar formatting conventions.

Thanks again for working on it!

**Comment by ahmedkrmn** [comment]
> The text looks good to me. Referencing grep, vgrep and rip-grep sounds like a good idea. Also, please have a look at the skopeo man page (https://raw.githubusercontent.com/containers/skopeo/master/docs/skopeo.1.md). We need the SYNOPSIS header and similar formatting conventions.
> 
> Thanks again for working on it!

Thanks Valentin,
I was unsure what to add in the "synopsis" section, should it be the same as `grep`'s synopsis?

**Comment by vrothberg** [comment]
> > The text looks good to me. Referencing grep, vgrep and rip-grep sounds like a good idea. Also, please have a look at the skopeo man page (https://raw.githubusercontent.com/containers/skopeo/master/docs/skopeo.1.md). We need the SYNOPSIS header and similar formatting conventions.
> > Thanks again for working on it!
> 
> Thanks Valentin,
> I was unsure what to add in the "synopsis" section, should it be the same as `grep`'s synopsis?

I think it should be:
```
vgrep [OPTION...] PATTERNS [FILE...]
vgrep [OPTION...] -s [COMMAND][CONTEXT] [SELECTORS]
```

**Comment by ahmedkrmn** [comment]
Please don't forget to add the "hacktoberfest-accepted" label to the PR!

File: docs/vgrep.1.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          commands: print, show, context, tree, delete, keep, refine, files, grep, quit, ?
+```
+
+vgrep commands can be passed directly to the `--show/-s` flag, for instance as `--show c5 1-10` to show the five context lines of the first ten matched lines. Furthermore, the commands can be executed in an interactive shell via the `--interactive/-i` flag. Running `vgrep --interactive` will enter the shell directly, `vgrep --show 1 --interactive` will first open the first matched line in the editor and enter the interactive shell after.
+
+## COMMANDS
+
+| Command | Description                                                                                                                                                                                                                                                                           |
+| ------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
```
Comment (vrothberg) [inline docs/vgrep.1.md:52]: Please delete this line. Otherwise, it does not render correctly in smaller terminal windows.

File: docs/vgrep.1.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+`vgrep` is inspired by the ancient **cgvg** scripts but extended to perform further operations such as listing statistics of files and directory trees or showing the context lines before and after the matches.
+
+`vgrep` runs on Linux, Windows and Mac OS.
+
+Note: `vgrep` is used to perform textual searches. On a technical level, vgrep serves as a front-end to grep or git-grep when invoking vgrep inside a git tree and uses `less` for displaying the results. All non-vgrep flags and arguments will be passed down to grep. Results of the last search are cached, so running vgrep without a new query will load previous results and operate on them.
+
+By default, the output will be written to less to make browsing large amounts of data more comfortable. vgrep --no-less will write to stdout.
+
+# Opening Matches
```
Comment (vrothberg) [inline docs/vgrep.1.md:23]:
```suggestion
## Opening Matches
```

File: docs/vgrep.1.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+```
+# export EDITOR=gedit
+# vgrep --show 4
+```
+
+The default editor of vgrep is `vim` with the default flag to open a file at a specific line being `+` followed by the line number. If your editor of choice hits the rare case of a different syntax, use the `EDITORLINEFLAG` environment variable to adjust. For example, a `kate` user may set the environment to `EDITOR="kate"` and `EDITORLINEFLAG="-l"`.
+
+Note that `vgrep` does not allow for searching and opening files at the same time. For instance, `vgrep --show=files text` should be split in two commands: `vgrep text` and `vgrep --show=files`.
+
+# Interactive Shell
```
Comment (vrothberg) [inline docs/vgrep.1.md:36]:
```suggestion
## Interactive Shell
```

File: docs/vgrep.1.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
+## NAME
```
Comment (vrothberg) [inline docs/vgrep.1.md:1]:
```suggestion
% VGREP(1) 

## NAME
```

**Comment by vrothberg** [comment]
Last minor change requests. After that we can merge :)

**Comment by vrothberg** [comment]
Friendly ping, @ahmedkrmn. Let's get this in :+1:

File: docs/vgrep.1.md (hunk 2 above)
Comment (ahmedkrmn) [reply docs/vgrep.1.md:52]: But that would remove the table, do you want it to be listed as bullet points?
I tested it on a small terminal window and it's rendered fine.
Capture

**Comment by vrothberg** [comment]
Thanks for working on this, @ahmedkrmn

**Pull request merged**